include/boost/hana/equal.hpp

```diff
@@ -191,7 +191,10 @@ BOOST_HANA_NAMESPACE_BEGIN
     }
 
     template <typename S>
-    struct equal_impl<S, S, when<hana::Struct<S>::value>> {
+    struct equal_impl<S, S, when<
+        hana::Struct<S>::value &&
+        !detail::EqualityComparable<S, S>::value
```

Comment:
Can you please add a test for that thing specifically? You could add it to `test/concept/struct/equal.cpp`.

Reply:
Actually, nevermind that. You already have one test for that in `github_460.cpp`.